fvtest/compilertriltest/CompareTest.cpp

```diff
@@ -1422,3 +1422,249 @@ INSTANTIATE_TEST_CASE_P(CompareTest, DoubleCompareOrUnordered, ::testing::Combin
         std::make_tuple<const char*, int32_t (*)(double, double)>("dcmpleu", dcmpleu)
 
     )));
+
+int32_t iffcmpequ(float l, float r) {
+    if (std::isnan(l)) return IFCMP_TRUE_NUM;
+    if (std::isnan(r)) return IFCMP_TRUE_NUM;
+    return (l == r) ? IFCMP_TRUE_NUM : IFCMP_FALSE_NUM;
+}
+
+int32_t iffcmpneu(float l, float r) {
+    if (std::isnan(l)) return IFCMP_TRUE_NUM;
+    if (std::isnan(r)) return IFCMP_TRUE_NUM;
+    return (l != r) ? IFCMP_TRUE_NUM : IFCMP_FALSE_NUM;
+}
+
+int32_t iffcmpltu(float l, float r) {
+    if (std::isnan(l)) return IFCMP_TRUE_NUM;
+    if (std::isnan(r)) return IFCMP_TRUE_NUM;
+    return (l < r) ? IFCMP_TRUE_NUM : IFCMP_FALSE_NUM;
+}
+
+int32_t iffcmpleu(float l, float r) {
+    if (std::isnan(l)) return IFCMP_TRUE_NUM;
+    if (std::isnan(r)) return IFCMP_TRUE_NUM;
+    return (l <= r) ? IFCMP_TRUE_NUM : IFCMP_FALSE_NUM;
+}
+
+int32_t iffcmpgeu(float l, float r) {
+    if (std::isnan(l)) return IFCMP_TRUE_NUM;
+    if (std::isnan(r)) return IFCMP_TRUE_NUM;
+    return (l >= r) ? IFCMP_TRUE_NUM : IFCMP_FALSE_NUM;
+}
+
+int32_t iffcmpgtu(float l, float r) {
+    if (std::isnan(l)) return IFCMP_TRUE_NUM;
+    if (std::isnan(r)) return IFCMP_TRUE_NUM;
+    return (l > r) ? IFCMP_TRUE_NUM : IFCMP_FALSE_NUM;
+}
+
+class FloatIfCompareOrUnordered : public TRTest::OpCodeTest<int32_t, float, float> {};
+
+TEST_P(FloatIfCompareOrUnordered, UsingConst) {
+    SKIP_ON_RISCV(MissingImplementation);
+    SKIP_ON_HAMMER(KnownBug) << "The x86_64 code generator crashes (see issue #5122)";
+
+    auto param = TRTest::to_struct(GetParam());
+
+    if ( param.opcode == "iffcmpneu" && (std::isnan(param.lhs) || std::isnan(param.rhs)) ) {
+        SKIP_ON_POWER(KnownBug) << "iffcmpneu returns wrong value on POWER (see #5152)";
+    }
```

Comment:
Only the `ne` variants should be affected by #5152 not the `neu` variants, so these checks should be removed.